File: src/EventManagement.Web/src/features/events/types/event.ts
/**
 * Lightweight event projection returned by the list and create endpoints. Mirrors the
 * backend's `EventDto`. Dates are ISO 8601 strings (UTC) and converted to the user's
 * timezone at render time by `formatDate`.
 */
export interface EventSummary {
  /** Stable event identifier (GUID string). */
  id: string;
  /** Display title. */
  title: string;
  /** Event date/time as ISO 8601, UTC. */
  date: string;
  /** Maximum number of registrations the event accepts. */
  maxCapacity: number;
  /** Current registration count at the time of the response. */
  currentRegistrations: number;
  /** When the event was created, ISO 8601 UTC. */
  createdAt: string;
}

/**
 * Full event projection returned by the detail and update endpoints. Mirrors the backend's
 * `EventDetailDto` and extends {@link EventSummary} with description and update timestamp.
 */
export interface EventDetail extends EventSummary {
  /** Optional long-form description. `null` when not set. */
  description?: string | null;
  /** When the event was last modified, ISO 8601 UTC. */
  updatedAt: string;
}

/**
 * Local form state used by `EventForm`. The `date` field carries the value from
 * `<input type="datetime-local">` (local time, no zone) and is converted to a UTC ISO
 * string at the submit boundary via `fromLocalInputValue`.
 */
export interface EventFormValues {
  title: string;
  description: string;
  /** `<input type="datetime-local">` value, e.g. `"2026-06-01T18:30"`. */
  date: string;
  maxCapacity: number;
}

File: src/EventManagement.Domain/Entities/Event.cs
namespace EventManagement.Domain.Entities;

/// <summary>
/// An event that users can register to attend. Owned by the Domain layer; mutated only through
/// the Application services so business invariants stay enforceable in one place.
/// </summary>
public sealed class Event
{
    /// <summary>Stable identifier for the event. Assigned on creation and never changes.</summary>
    public Guid Id { get; init; }

    /// <summary>Human-readable title shown in listings and on the event detail page.</summary>
    public string Title { get; set; } = string.Empty;

    /// <summary>Optional long-form description. May be null when no description is supplied.</summary>
    public string? Description { get; set; }

    /// <summary>Date and time the event takes place. Stored as UTC; the UI converts to the user's local zone at the boundary.</summary>
    public DateTimeOffset Date { get; set; }

    /// <summary>Maximum number of registrations the event will accept. Enforced by <see cref="Services.RegistrationRules"/>.</summary>
    public int MaxCapacity { get; set; }

    /// <summary>Timestamp the event row was first created. Set once and never modified.</summary>
    public DateTimeOffset CreatedAt { get; init; }

    /// <summary>Timestamp of the most recent update to the event.</summary>
    public DateTimeOffset UpdatedAt { get; set; }
}
